src/index.ts

```diff
@@ -216,8 +216,7 @@ type PathValueEntry<T, P extends PathEntry<T, Depth>, Depth extends number = 10>
   BuildTuple<Depth>
 >
 
-type SafeObject = Record<string, any>
-type SearchableObject = Record<never, never> | unknown[]
+type SearchableObject = Record<string | number, unknown>
```

Comment:
@djfhe you used never, never here explicitly in a commit (https://github.com/clickbar/dot-diver/commit/1e230f61ab7250fd69e2aecc1f1adcd32544d500) 
Do you think this is fine?

Otherwise i think `Record<string | number | symbol, unknown>` is even more correct and should be used, since this type is also used for the first parameter of the `setByPath` method for example.
 
 
 One issue I can think of would be, that we now do not allow arrays as top level elements anymore.
 Atm i do not know for sure if the code even worked for these cases, but the typing suggests it does.

Reply:
You're absolutely right, I pushed a fix for that, as for the never as Record key, typescript already only allows symbol string or numbers as keys, so `never` doesn't change anything externally, it only makes indexing fail which we use a lot internally

![image](https://github.com/clickbar/dot-diver/assets/6115458/5ac41dde-ad5f-4460-9ad0-bf9e024b411e)